src/Form/index.tsx

```diff
@@ -329,6 +332,39 @@ function Form({
     );
   }, [activeStep?.id]);
 
+  useEffect(() => {
+    if (
+      !hasPopupBeenShown &&
+      popupOptions !== undefined &&
+      popupOptions.show === true
+    ) {
+      // We don't want to run the cleanup logic below until the popup has been shown
+      setHasPopupBeenShown(true);
+    } else if (
+      hasPopupBeenShown &&
+      popupOptions !== undefined &&
+      popupOptions.show === false &&
+      !initInfo().formSessions[formName].track_users
+    ) {
+      const wipedValues = { ...fieldValues };
+      // @ts-expect-error
+      Object.keys(wipedValues).forEach((key) => (wipedValues[key] = undefined));
+      updateFieldValues(wipedValues);
+      initState.userId = uuidv4();
+      history.replace(location.pathname + location.search);
+      const newKey = getInitialStep({
+        initialStepId,
+        steps,
+        sessionCurrentStep: session.current_step_key
+      });
+      // if I set this, then the hash shows while the form is closed. If I don't
+      // set this, the hash stays cleared until the form is both reopened &
+      // navigated to a new step to another step
+      // setUrlStepHash(history, steps, newKey);
```

Comment:
is it fine if the hash doesn't show up until navigating to the 2nd step in this case? happy to adjust this any way - but wasn't sure what you'd expect the bx to be